### PR TITLE
Repermission cookie check to access consent page

### PR DESCRIPTION
**Pull request by Calum-Campbell**
## What does this change?
- Allows access to the /consent page if user has a valid SU_GU_RP cookie or a recently authenticated SC_GU_LA cookie (same level needed to access /profile pages).
- Related to https://github.com/guardian/identity/pull/731 .
- Tests failing as the identity-cookie library has not been published yet.

## Tested in CODE?
No

### CODE REVIEW

File: identity/app/actions/AuthenticatedActions.scala [hunk 1]
```diff
@@ -67,6 +67,19 @@ class AuthenticatedActions(
       }
   }
 
+  def permissionRefiner: ActionRefiner[Request, AuthRequest] = new ActionRefiner[Request, AuthRequest] {
+    override val executionContext = ec
+
+    def refine[A](request: Request[A]) =
+      authService.authenticateUserForPermissions(request) match {
+        case Some(permUser) => Future.successful(Right( new AuthenticatedRequest( permUser, request)))
```
Comment (jfsoul) [inline identity/app/actions/AuthenticatedActions.scala]: Unusual spacing on this line.  You need to sort out your auto-formatting settings in Intellij 😄

**Review by jfsoul — APPROVED** [review]
Looks good!

File: identity/app/services/AuthenticationService.scala [hunk 2]
```diff
@@ -25,7 +25,9 @@ class AuthenticationService(cookieDecoder: FrontendIdentityCookieDecoder,
     minimalSecureUser <- cookieDecoder.getUserDataForScGuU(scGuU.value)
     guUCookieData <- cookieDecoder.getUserDataForGuU(guU.value)
     fullUser = guUCookieData.getUser if (fullUser.getId == minimalSecureUser.getId)
-  } yield AuthenticatedUser(fullUser, ScGuU(scGuU.value, guUCookieData))
+    scGuLa <- request.cookies.get("SC_GU_LA")
+    hasRecentlyAuthenticated = cookieDecoder.userHasRecentScGuLaCookie(fullUser, scGuLa.value, Minutes.minutes(20).toStandardDuration)
+  } yield AuthenticatedUser(fullUser, ScGuU(scGuU.value, guUCookieData), hasRecentlyAuthenticated)
 
   def recentlyAuthenticated(request: RequestHeader): Boolean = (for {
```
Comment (jfsoul) [inline identity/app/services/AuthenticationService.scala]: Probably preferable to remove this entirely and update calling code.

File: identity/app/services/AuthenticationService.scala [hunk 3]
```diff
@@ -25,7 +25,9 @@ class AuthenticationService(cookieDecoder: FrontendIdentityCookieDecoder,
     minimalSecureUser <- cookieDecoder.getUserDataForScGuU(scGuU.value)
     guUCookieData <- cookieDecoder.getUserDataForGuU(guU.value)
     fullUser = guUCookieData.getUser if (fullUser.getId == minimalSecureUser.getId)
-  } yield AuthenticatedUser(fullUser, ScGuU(scGuU.value, guUCookieData))
+    scGuLa <- request.cookies.get("SC_GU_LA")
```
Comment (jfsoul) [inline identity/app/services/AuthenticationService.scala]: So in the case where a user has all valid cookies, but is missing "SC_GU_LA" then this for comprehension returns `None`?  I don't think that's the intention?

File: identity/app/services/AuthenticationService.scala (hunk 2 above)
Comment (Calum-Campbell) [reply identity/app/services/AuthenticationService.scala]: Yep good shout

File: identity/app/services/AuthenticationService.scala (hunk 3 above)
Comment (Calum-Campbell) [reply identity/app/services/AuthenticationService.scala]: ah you're right, something like  : 
```
 def authenticatedUserFor[A](request: RequestHeader): Option[AuthenticatedUser] = for {
    scGuU <- request.cookies.get("SC_GU_U")
    guU <- request.cookies.get("GU_U")
    minimalSecureUser <- cookieDecoder.getUserDataForScGuU(scGuU.value)
    guUCookieData <- cookieDecoder.getUserDataForGuU(guU.value)
    fullUser = guUCookieData.getUser if (fullUser.getId == minimalSecureUser.getId)
  } yield {
    val hasRecentlyAuthenticated = request.cookies.get("SC_GU_LA").map(scgula => cookieDecoder.userHasRecentScGuLaCookie(fullUser,scgula.value, Minutes.minutes(20).toStandardDuration))
    AuthenticatedUser(fullUser, ScGuU(scGuU.value, guUCookieData), hasRecentlyAuthenticated.getOrElse(false))
  }
```
??

**Review by jfsoul — APPROVED** [review]

**Pull request merged**